lib/timeutils/format.c

```diff
@@ -46,13 +46,13 @@ _append_frac_digits(glong usecs, GString *target, gint frac_digits)
     }
 }
 
-static gint
-_format_zone_info(gchar *buf, size_t buflen, glong gmtoff)
+void
+append_format_zone_info(GString *target, glong gmtoff)
```

Comment:
Are you sure this is performance wise better? The original one time append is replaced with four appends, which can result in four reallocs. Also, format_uint32_padded contains a lot of function calls and parse logic.

I am asking, because the original code was way more readable.

Reply:
the original was using snprintf() which is _slow_ (for a fastpath), that's why format_uint32_padded() and friends were born in the first place.

Also, GString instances are reallocated in exponential terms, so a single realloc will probably suffice even if the GString instance doesn't have enough space at the start of the function. But it usually would have that space, as at this call we are usually appending the timezone information at the end of the timestamp (19 characters into the string, which would have been set to 32 characters when we were at 16).

As for readability, I agree. snprintf() just doesn't cut it performance wise, at least in our fastpath. If the user sets a ISOTIME in his/her template or uses RFC5424, this can make a difference.

With all that said, I didn't do a detailed micro-benchmark of this change, but as you can see, we are already using format_uint32_padded() functions in the same file for these reasons.

And even if sprintf() has improved since this was implemented, I still would retain to the optimized implementation, simply for platform independence.